common/lib/xmodule/xmodule/modulestore/split_mongo/split_draft.py

```diff
@@ -189,49 +189,37 @@ def delete_item(self, location, user_id, revision=None, skip_auto_publish=False,
                     currently only provided by contentstore.views.item.orphan_handler
                 Otherwise, raises a ValueError.
         """
+        allowed_revisions = [
+            None,
+            ModuleStoreEnum.RevisionOption.published_only,
+            ModuleStoreEnum.RevisionOption.all
+        ]
+        if revision not in (allowed_revisions):
+            raise UnsupportedRevisionError(allowed_revisions)
+
+        autopublish_parent = False
         with self.bulk_operations(location.course_key):
             if isinstance(location, LibraryUsageLocator):
                 branches_to_delete = [ModuleStoreEnum.BranchName.library]  # Libraries don't yet have draft/publish support
-            elif revision == ModuleStoreEnum.RevisionOption.published_only:
-                branches_to_delete = [ModuleStoreEnum.BranchName.published]
+            elif location.category in DIRECT_ONLY_CATEGORIES:
+                branches_to_delete = [ModuleStoreEnum.BranchName.published, ModuleStoreEnum.BranchName.draft]
             elif revision == ModuleStoreEnum.RevisionOption.all:
                 branches_to_delete = [ModuleStoreEnum.BranchName.published, ModuleStoreEnum.BranchName.draft]
-            elif revision is None:
-                branches_to_delete = [ModuleStoreEnum.BranchName.draft]
-                draft_location = location.for_branch(ModuleStoreEnum.BranchName.draft)
-                try:
-                    item = self.get_item(draft_location)
-                    if getattr(item, 'has_children', False):
-                        # If item have has_published_version then delete published children also.
-                        if self.has_published_version(item):
-                            branches_to_delete.insert(0, ModuleStoreEnum.BranchName.published)
-                except ItemNotFoundError:
-                    # Raises ValueError as in function description
-                    raise ValueError("Cannot delete a block that does not exist")
             else:
-                raise UnsupportedRevisionError(
-                    [
-                        None,
-                        ModuleStoreEnum.RevisionOption.published_only,
-                        ModuleStoreEnum.RevisionOption.all
-                    ]
-                )
+                if revision == ModuleStoreEnum.RevisionOption.published_only:
+                    branches_to_delete = [ModuleStoreEnum.BranchName.published]
+                elif revision is None:
+                    branches_to_delete = [ModuleStoreEnum.BranchName.draft]
+                    parent_loc = self.get_parent_location(location.for_branch(ModuleStoreEnum.BranchName.draft))
+                    autopublish_parent = not skip_auto_publish and parent_loc.block_type in DIRECT_ONLY_CATEGORIES
```

Comment:
See my other note for the test that a draft-deleted child of a DOC is removed from both published and draft branches.